### ISSUE TITLE
Which versions work together?

### ISSUE THREAD

**Issue by KyleGobel** [Azure/autorest.typescript#136]
I'm having a hell of a time trying to find versions that work together, between this and `ms-rest-js`.

**Out of the box:**
ms-rest-js: 0.4.0
autorest-core: 2.0.4279
autorest.typescript: (~2.0.12->2.0.179) ??? (Not sure what version this is supposed to mean.  Using whatever it picked up when i type `autorest ./readme.md` by default.

Issue here is that `ms-rest-js` contains no URLBuilder (that apparently came in later versions).

**Upgrading to ms-rest-js preview:**
ms-rest-js: 0.7.224 (published two hours ago..so maybe too* new)
autorest-core: 2.0.4279
autorest.typescript: (~2.0.12->2.0.179)

With this we get the error that was reported in #122 (`{} is not assignable to type HttpHeaders`)

**Upgrading this package (autorest.typescript) to the latest:**
ms-rest-js: 0.7.224
autorest-core: 2.0.4279
autorest.typescript: 2.0.282

(now running with `autorest ./README.md --use=@microsoft.azure/autorest.typescript@2.0.282`)

This one actually compiles fine but running requests don't work (unless this problem is on my end).

Drilled down to find the requests are running fine, but there is a section in the template that is doing this:
```typescript
  // Deserialize Response
      if (statusCode === 200) {
        let parsedResponse = operationRes.parsedBody as { [key: string]: any };
      //...
```
`parsedBody` is undefined for me...not sure where this is expected to be set.  The `bodyAsText` though is perfectly fine json.

So all calls i've tried always return `undefined`.

Wondering what set of versions I can use where I can get something working.  Any tips or pointers are appreciated.

**Comment by KyleGobel** [comment]
The latest problem I think is with with ms-rest-js.  I filed an issue on that repo.

https://github.com/Azure/ms-rest-js/issues/120

**Comment by KyleGobel** [comment]
If it helps anyone, a working combination would be:

**ms-rest-js**: 0.7.227 (newer version contains some name changes that breaks this)
**autorest.typescript**: 2.0.282

**Comment by RikkiGibson** [comment]
Thanks for bringing up this pretty glaring usability problem. We're thinking of adding the expected ms-rest-js and ms-rest-azure-js version for each autorest.typescript version to the README or possibly having the generator process log those versions at generation time so you can see them.

**Comment by chaholl** [comment]
I've also been wrestling with this problem for most of today. The versions suggested by @KyleGobel doesn't seem to work for me. (Although it's fair to say that they yield far fewer errors than any of the other combinations I've tried so far).

Can anybody advise of a working combination? Or should I call it a day and switch to Swagger-Codegen?

**Comment by daschult** [comment]
We just promoted the most recent preview versions of autorest.typescript, ms-rest-azure-js, and ms-rest-js to be the latest versions. If you try to run AutoRest with the --typescript parameter and no version numbers, it should just work at this point.

**Comment by daschult** [comment]
@chaholl @KyleGobel Are you guys able to generate working code with the latest versions of the generator?

**Comment by chaholl** [comment]
Thanks @daschult 
Unfortunately still doesn't work for me. If I remove the -use=@microsoft.azure/autorest.typescript@2.0.282_
option it reverts to 2.0.157 and throws all kinds of errors.

I'm using: 
- autorest  2.04262
- autorest.typescript  2.0.157
- autorest.modeler  2.1.23
- ms-rest-js   0.8.235
- typescript   2.4.1

Some example errors:
`` TS7017: Element implicitly has an 'any' type because type 'HttpHeaders' has no index signature.
TS2339: Property 'response' does not exist on type 'HttpOperationResponse<any, any>'.

**Comment by daschult** [comment]
Yeah, AutoRest can be a big pain with specifying specific extension versions to use.
I recommend calling `autorest --reset` and then running `autorest --typescript` to ensure that you get the latest version of the AutoRest TypeScript extension (should be [2.0.304](https://www.npmjs.com/package/@microsoft.azure/autorest.typescript/v/2.0.304)).

**Comment by chaholl** [comment]
Thanks again @daschult 
Resetting did pick up the latest version and I found that I also had to update typescript to 2.9.1. After that everything worked as expected. So just to confirm for anybody else:

Package | Version
------------ | -------------
autorest | 2.04262
autorest.typescript | 2.0.304
autorest.modeler | 2.1.23
ms-rest-js | 0.8.235
typescript | 2.9.1

I also found that setting 'strict': to false  in tsconfig.json was required to get all this working with React but that's possibly unrelated

**Comment by daschult** [comment]
Thanks for the table, @chaholl. We're trying to make this product and extension as easy to work with as possible, so your feedback is always appreciated.

**Comment by daschult** [comment]
@KyleGobel Since @chaholl was able to get this work, I'm going to assume you can too. If you're still having problems, please reopen this issue and reach out to me. I'll be happy to help figure out whatever problem you're running it.

**Comment by RikkiGibson** [comment]
@daschult It would be good for us to note which TypeScript version we require. I assume that there aren't usually many pains associated with migrating to latest TypeScript. I wonder which recent features we were using in the gen code that required that upgrade? Hard to keep track.

**Comment by daschult** [comment]
I've updated the README.md file for this repository to indicate the versions of ms-rest-js and ms-rest-azure-js that the specific version of the generator targets. These values are auto-generated, so they should always stay up to date.
